src/main/scala/Schema.scala

```diff
@@ -0,0 +1,62 @@
+package CSVValidation
+
+import com.fasterxml.jackson.databind.{JsonNode, ObjectMapper}
+import com.fasterxml.jackson.databind.node.ObjectNode
+import traits.JavaIteratorExtensions.IteratorHasAsScalaArray
+
+import java.net.URL
+import java.nio.file.Paths
+object Schema {
+  val objectMapper = new ObjectMapper()
+
+  def loadMetadataAndValidate(
+      uri: String,
+      outputErrors: Boolean = true
+  ): (Option[TableGroup], Option[String]) = {
+    try {
+      val jsonNode = objectMapper.readTree(Paths.get(uri).toFile)
+      (
+        Some(
+          Schema
+            .fromCsvwMetadata(s"file:$uri", jsonNode.asInstanceOf[ObjectNode])
+        ),
+        None
+      )
+    } catch {
+      case metadataError: MetadataError => {
+        (None, Some(metadataError.getMessage))
+      }
+      case _: Throwable => {
+        (None, Some("Unknown Exception occurred"))
```

Comment:
I think it'd be a good idea to keep the contents of the exception and print it out to the console. It can really help when debugging.

Reply:
The stacktrace is good to output too.

Reply:
okay, changed it to return exception message and stack-trace.